app/src/test/java/com/example/trainingintive/dog_images_feature/data/repository/DogImageRepositoryImplTest.kt

```diff
@@ -0,0 +1,177 @@
+import com.example.trainingintive.dog_images_feature.domain.model.DogImageUrl
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.unmockkStatic
+import io.mockk.verify
+import io.reactivex.rxjava3.core.Completable
+import io.reactivex.rxjava3.core.Flowable
+import io.reactivex.rxjava3.core.Single
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import java.util.Collections
+
+class DogImageRepositoryImplTest {
+
+    val dogApiService: DogImageApiService = mockk()
+    val dogDao: DogImageDao = mockk()
+    val tested = DogImageRepositoryImpl(dogApiService, dogDao)
+    val urlApiModel: DogImageUrlApiModel = mockk(relaxed = true)
+    val urlDomain: DogImageUrl = mockk(relaxed = true)
+
+    @Before
+    fun start() {
+        mockkStatic(DogImageUrl::toDatabaseEntity)
+    }
+
+    @After
+    fun finish() {
+        unmockkStatic(DogImageUrl::toDatabaseEntity)
+    }
+
+    @Test
+    fun `api service should be called when getting url`() {
+        val amountUrlsInDb = 1
+        every { dogApiService.getDogImage() } returns Single.just(urlApiModel)
+        every { dogDao.count() } returns Single.just(amountUrlsInDb)
+
+        tested.getDogImageUrl().test()
+
+        verify { dogApiService.getDogImage() }
+    }
+
+    @Test
+    fun `count method should be called when getting url`() {
+        val amountUrlsInDb = 1
+        every { dogApiService.getDogImage() } returns Single.just(urlApiModel)
+        every { dogDao.count() } returns Single.just(amountUrlsInDb)
+
+        tested.getDogImageUrl()
+
+        verify { dogDao.count() }
+    }
+
+    @Test
+    fun `get dog image url`() {
+        val amountUrlsInDb = 1
+        every { dogApiService.getDogImage() } returns Single.just(urlApiModel)
+        every { dogDao.count() } returns Single.just(amountUrlsInDb)
+        val expected = urlApiModel.toDomain(amountUrlsInDb)
+
+        val result = tested.getDogImageUrl().test()
+
+        result.assertValue(expected)
+    }
+
+    @Test
+    fun `get all urls should return list of domain models`() {
+        val dbModels: List<DogImageUrlDb> = listOf(
+            mockk(relaxed = true),
+            mockk(relaxed = true),
+            mockk(relaxed = true),
+        )
+        every { dogDao.getAllImageUrls() } returns Flowable.just(dbModels)
+        val expected = dbModels.map { it.toDomain() }
+
+        val result = tested.getAllImageUrls().test()
+
+        result.assertValue(expected)
+    }
+
+    @Test
+    fun `dao should be called when getting all urls`() {
+        every { dogDao.getAllImageUrls() } returns Flowable.just(emptyList())
+
+        tested.getAllImageUrls()
+
+        verify { dogDao.getAllImageUrls() }
+    }
+
+    @Test
+    fun `test insert`() {
+        every { dogDao.insert(any()) } returns Completable.complete()
+        every { urlDomain.toDatabaseEntity() } returns mockk(relaxed = true)
+        val valueToInsert = urlDomain.toDatabaseEntity()
```

Comment:
Create `val valueToInsert = mockk()` at the top of the file and return it from `every { urlDomain.toDatabaseEntity() } returns valueToInsert`. Same in other places